projects/convai2/baselines/seq2seq/eval_ppl.py

```diff
@@ -59,8 +59,7 @@ def next_word_probability(self, partial_out):
             prev_enc=self.prev_enc)
         scores, self.prev_enc = out[1], out[-1]
         # scores is bsz x seqlen x num_words, so select probs of current index
-        assert len(partial_out) == scores.size(1) - 1
-        probs = F.softmax(scores.select(1, len(partial_out)), dim=1).squeeze()
+        probs = F.softmax(scores.select(1, -1), dim=1).squeeze()
```

Comment:
what changed here?

Reply:
basically: you can use a different tokenizer if you want, but the last token you produce should be the one you're predicting. before it was using the last token, but you had to be using the exact same tokenizer.